src/development/metrics.md

```diff
@@ -0,0 +1,153 @@
+# Metrics
+
+This page gather all the measurement metrics, so that we can easily refer to them in the future. It is broken into four parts:
+
+ - The measurements that are common to both verkle and binary trees
+ - The measurements that are specific to verkle trees
+ - The measurements that are specific to binary trees
+ - The measurements that are specific to state expiry
+
+This is a living document, that we will fill as questions arise and we gather more data.
+
+##  Common measurements
+
+### Execution performance
+
+#### impact of key hashing
+
+In order to estimate the performance impact of Pedersen key hashing, this is a comparison of how long it takes to replay 200k blocks using Pedersen hashes, vs. sha256:
+
+![image.png](./assets/hash_speed_diff.png)
+
+At the end of the sha256 run, so >200k blocks later, the chain was 75072 blocks ahead. This represents a loss of 3 hours per day against sha256. That’s >12% faster.
+
+Coupled to the fact that Pedersen hashes aren’t quantum resistant, it seems like a good idea to reconsider using Pedersen hashes to compute trees.
+
+## Verkle-specific measurements
+
+### Context
+
+This data is gathered replaying ~200k historical blocks around the Shanghai fork. While they do not give a perfect view of how execution will behave once EIP-4762 is active, it still gives a good indication of what to expect, and how the spec should evolve.
+
+### Witnesses
+
+#### Size
+
+Research has been conducted, in order to determine the best structure for a witness. We tried three methods:
```

Comment:
It mentions there're "three methods" but I see four bullet points?
I also think interleaving bullet points with images is hard to read.

I'd suggest saying: "Three witness encoding formats where explored, which are described below."

Then:
**Type 0**
Specified in the [current version of the consensus spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip6800/beacon-chain.md#executionwitness), referred to as **type 0**. This approach ...

**Type 1 serialization container**
(Python def...)
 A first variation of the spec, in which prestate val....

**Type 2 serialization container**
(Python def...)
A second variation of the spec, in which update....

etc